library/pk.c

```diff
@@ -659,6 +659,9 @@ static int asn1_write_mpibuf( unsigned char **p, unsigned char *start,
         len += 1;
     }
 
+    if( ( *p - start ) < 2 )
```

Comment:
@AndrzejKurek,  please add comment for this check

Reply:
Done.